src/server/pfs/server/driver.go

```diff
@@ -1247,139 +1247,85 @@ func (d *driver) listCommit(ctx context.Context, repo *pfs.Repo, to *pfs.Commit,
 	return commitInfos, nil
 }
 
-type commitStream struct {
-	stream chan CommitEvent
-	done   chan struct{}
-}
-
-func (c *commitStream) Stream() <-chan CommitEvent {
-	return c.stream
-}
-
-func (c *commitStream) Close() {
-	close(c.done)
-}
-
-func (d *driver) subscribeCommit(ctx context.Context, repo *pfs.Repo, branch string, from *pfs.Commit) (CommitStream, error) {
+func (d *driver) subscribeCommit(ctx context.Context, repo *pfs.Repo, branch string, from *pfs.Commit, f func(*pfs.CommitInfo) error) error {
```

Comment:
It looks like there are a few places in the pfs proto where a string is being used instead of the pfs.Branch type. Is there a reason for this? I am commenting this here because not having a protobuff type for the branch threw me off here.